lib/rexml/xpath_parser.rb

```diff
@@ -78,7 +78,15 @@ def variables=( vars={} )
 
     def parse path, node
       path_stack = @parser.parse( path )
-      match( path_stack, node )
+      if node.is_a?(Array)
+        Kernel.warn("REXML::XPath.each, REXML::XPath.first, REXML::XPath.match dropped support for nodeset...", uplevel: 1)
+        return [] if node.empty?
+        node = node.first
+      end
+
+      node.document.__send__(:enable_cache) do
+        match( path_stack, node )
+      end
```

Comment:
Can we add a test for this cache?

For example:
1. It uses `XPathParser#parse`
2. It changes namespace
3. It uses `XPathParser#parse` (this should not use cache used for 1.)

Reply:
OK.
I see.